docs/TEST_GUIDE.md

```diff
@@ -0,0 +1,75 @@
+# Test guide
+
+This project uses 3 layers of automated tests (unit, integration, and end-to-end). 
+It is very powerful setup, but at the same time it can be challenging to choose appropriate 
+level of test for the problem at hand. 
+This is the reason why this guide was created. 
+
+The goal of this guide is to provide an overview of used types of tests, describe their pros and cons, and give hints for when to use which.
+
+## Unit tests
+
+### What do they test?
+One unit ([component](/LiskHQ/lisk-hub/blob/development/src/components),
+[util](/LiskHQ/lisk-hub/blob/development/src/utils),
+[action](/LiskHQ/lisk-hub/blob/development/src/actions),
+[reducer](/LiskHQ/lisk-hub/blob/development/src/store/reducers),
+[middleware](/LiskHQ/lisk-hub/blob/development/src/store/middlewares), or
+[subscriber](/LiskHQ/lisk-hub/blob/development/src/store/subscribers)) in isolation.
+
+### How are they organized?
+Each unit test live in the same folder as the unit that it tests. E.g. tests for [/src/components/login/login.js](/LiskHQ/lisk-hub/blob/development/src/components/login/login.js) are in [/src/components/login/login.test.js](/LiskHQ/lisk-hub/blob/development/src/components/login/login.test.js).
+
+Configuration is in [karma.conf.js](/LiskHQ/lisk-hub/blob/development/karma.conf.js) and [config/webpack.config.test.js](/LiskHQ/lisk-hub/blob/development/config/webpack.config.test.js).
+
+### How to run them?
+See [relevant sections of README](/LiskHQ/lisk-hub#run-unit-tests)
+
+### What tools are used?
+- **Assertions** use `expect` syntax of [chai](http://www.chaijs.com/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/80afda8289b75cb70bf345d14d36117fde2bdd90/src/utils/passphrase.test.js#L136
+
+- **Spies, stubs, ands mocks** use [sinon](http://sinonjs.org/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/19a2a613fa08a34fe3088b0e40c11c7fa37a645d/src/store/middlewares/savedSettings.test.js#L20
+
+- To test **React compoments** we use [enzyme](http://airbnb.io/enzyme/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/79165170a326a7f98efee098732e55be37d31223/src/components/toolbox/sliderCheckbox/index.test.js#L21
+
+
+## Integration tests
+
+### What do they test?
```

Comment:
I believe it is vital to note here to not test the same functionality which has been tested in unit tests and/or e2e tests. we have to test logics which may result in different behaviour/response if the `state` was different from our mock in unit tests. 
Take [transactionID](https://github.com/LiskHQ/lisk-hub/blob/0.4.0/test/integration/transactionID.test.js) for instance.
The [SingleTransaction](https://github.com/LiskHQ/lisk-hub/blob/0.4.0/src/components/singleTransaction/index.js) component contains a simple JSX which receives `transaction` and `peers` through `redux-connect`. We test the corresponsing action and reducer using unit tests, we "should" have an e2e test scenario starting from login page, through transaction list and ending in transaction detail page. So I can not see a  remarkable value added by this integration test.